### PR TITLE
Identify text fields as such to a11y

### PR DESCRIPTION
**Pull request by goderbauer**
Requires engine rolled to https://github.com/flutter/engine/pull/4299.

Required for: https://github.com/flutter/flutter/issues/12785, https://github.com/flutter/flutter/issues/12786.

### CODE REVIEW

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -250,4 +255,33 @@ void main() {
...
+
+    expect(semantics, includesNodeWith(flags: <SemanticsFlags>[SemanticsFlags.isTextField]));
+
+    await tester.tap(find.byType(EditableText));
+    await tester.idle();
+    await tester.pump();
+
+    expect(semantics, includesNodeWith(flags: <SemanticsFlags>[SemanticsFlags.isTextField, SemanticsFlags.isFocused]));
+
+  });
```
Comment (amirh) [inline packages/flutter/test/widgets/editable_text_test.dart]: Should we test the non focused case as well?

File: packages/flutter/test/widgets/semantics_tester.dart [hunk 2]
```diff
@@ -362,33 +372,33 @@ class _IncludesNodeWith extends Matcher {
...
-      string += 'actions "${actions.join(', ')}"';
-    }
-    return string;
+    final List<String> strings = <String>[];
+    if (label != null)
+      strings.add('label "$label"');
+    if (textDirection != null)
+      strings.add(' (${describeEnum(textDirection)})');
+    if (actions != null)
+    strings.add('actions "${actions.join(', ')}"');
```
Comment (amirh) [inline packages/flutter/test/widgets/semantics_tester.dart]: indent

File: packages/flutter/test/widgets/semantics_tester.dart [hunk 3]
```diff
@@ -362,33 +372,33 @@ class _IncludesNodeWith extends Matcher {
...
+    if (flags != null)
+    strings.add('flags "${flags.join(', ')}"');
```
Comment (amirh) [inline packages/flutter/test/widgets/semantics_tester.dart]: indent

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/test/widgets/editable_text_test.dart]: Isn't that covered by line 278?

Comment (amirh) [reply packages/flutter/test/widgets/editable_text_test.dart]: oops

**Review by amirh — APPROVED** [review]
LGTM

**Pull request merged**